Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import pandas as pd 

df =  pd.DataFrame({
        "hrs_of_study": [10,8,11,6,9],
        "score": [170,155,180,140,160]
})

correlation = df.corr()

correlation

hrs_of_study     score
hrs_of_study      1.000000  0.994107
score             0.994107  1.000000

Interpreting the result:

The correlation matrix shows the correlation coefficients between the variables "hrs_of_study" and "score". Both variables have a correlation coefficient of approximately 0.986,This implies that there is a strong positive association between the two variables. In other words, students who spend more hours studying tend to achieve higher scores on their exams.

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.

In [2]:
# creating the data 
data = {
    "amount_of_sleep": [8,7,6,5,6,7],
    "job_satisfaction": [7,5,8,9,8,7]
}

# creating df from data
df = pd.DataFrame(data)

# calcuating spearman corelation rank
spearman_corr = df["job_satisfaction"].corr(df["amount_of_sleep"],method="spearman")

# printing the output
spearman_corr

-0.8636363636363634

Interpreting the result:


The coefficient between "amount_of_sleep" and "job_satisfaction" is approximately -0.8636. 

A negative correlation coefficient indicates that as the amount of sleep decreases, the job satisfaction level tends to decrease as well, and vice versa. However, it's important to note that the correlation coefficient of -0.863636 suggests a strong relationship, meaning that the relationship is particularly strongly negetive .

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.

In [3]:
# creating data 
data = {
    "exercise_hours": [3, 5, 4, 2, 6, 1, 4, 3, 2, 5, 6, 3, 2, 4, 1, 5, 2, 3, 4, 6, 2, 1, 3, 4, 2,
                  3, 5, 4, 2, 6, 1, 4, 3, 2, 5, 6, 3, 2, 4, 1, 5, 2, 3, 4, 6, 2, 1, 3, 4, 2],
    
    "bmi": [22.5, 24.3, 26.1, 21.9, 28.6, 20.4, 25.8, 23.7, 22.1, 24.9, 26.8, 23.1, 21.6, 25.1, 20.1,
           24.7, 21.5, 23.4, 25.6, 27.3, 21.3, 19.8, 23.0, 24.5, 21.8, 23.2, 24.8, 26.2, 22.9, 23.6,
           27.1, 20.6, 24.2, 22.8, 25.7, 26.5, 23.3, 21.7, 25.3, 20.9, 23.9, 26.0, 27.4, 22.3, 20.2,
           23.8, 25.0, 20.5, 22.7, 24.6]
}

# creating df from data
df = pd.DataFrame(data)

# Calculate the Pearson correlation coefficient
pearson_corr = df['exercise_hours'].corr(df['bmi'])

# Calculate the Spearman's rank correlation coefficient
spearman_corr = df['exercise_hours'].corr(df['bmi'], method='spearman')

# Print the correlation coefficients
print("Pearson correlation coefficient:", pearson_corr)
print("Spearman's rank correlation coefficient:", spearman_corr)

Pearson correlation coefficient: 0.48979201270803113
Spearman's rank correlation coefficient: 0.49488487431783923


Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.

In [4]:
import numpy as np

#creating random saple data using numpy 
data = {
    "tv_hours": np.random.randint(0, 5, size=50),
    "physical_activity": np.random.randint(0, 5, size=50)

}

# Create the dataset
df = pd.DataFrame(data)

# Calculate the Spearman's rank correlation coefficient
pearson_corr = df["tv_hours"].corr(df["physical_activity"])

# Print the correlation coefficients
print("Pearson correlation coefficient:", pearson_corr)

Pearson correlation coefficient: 0.17665116104425108


Q5. A survey was conducted to examine the relationship between age and preference for a particular brand of soft drink. The survey results are shown below:

    Age(Years)        Soft drink Preference
    25                Coke
    42                Pepsi
    37                Mountain dew
    19                Coke
    31                Pepsi
    28                Coke

In [5]:
import pandas as pd

# Create a DataFrame with the survey results
data = {
    "Age(Years)": [25, 42, 37, 19, 31, 28],
    "Soft_drink_Preference": ["Coke", "Pepsi", "Mountain Dew", "Coke", "Pepsi", "Coke"]
}

df = pd.DataFrame(data)

# Perform label encoding on the "Soft drink Preference" column
label_mapping = {"Coke": 0, "Pepsi": 1, "Mountain Dew": 2}

df["Soft_drink_Preference"] = df["Soft_drink_Preference"].map(label_mapping)

# Calculate the Pearson correlation coefficient
pearson_corr = df["Age(Years)"].corr(df["Soft_drink_Preference"])

# Print the correlation coefficient
print("Pearson correlation coefficient:", pearson_corr)


Pearson correlation coefficient: 0.7587035441865058


Q6. A company is interested in examining the relationship between the number of sales calls made per day and the number of sales made per week. The company collected data on both variables from a sample of 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [6]:
data = {
"sales_calls_per_day": [10, 15, 12, 8, 14, 20, 18, 10, 16, 9, 11, 13,15, 17, 10, 
                       16, 19, 20, 22, 14, 12, 11, 15, 17,12, 14, 20, 18, 13, 15],

"sales_per_week" : [2, 3, 2, 1, 3, 4, 3, 2, 3, 1, 2, 2, 3, 3, 4, 
                   2, 2, 3, 3, 4, 4, 3, 2, 2, 3, 3, 4, 3, 2, 3]
}

df = pd.DataFrame(data)

pearson_corr = df["sales_calls_per_day"].corr(df["sales_per_week"])

print("Pearson correlation coefficient:", pearson_corr)

Pearson correlation coefficient: 0.41281208793920454
